lib/bundler/shared_helpers.rb

```diff
@@ -336,7 +336,11 @@ def clean_load_path
       loaded_gem_paths = Bundler.rubygems.loaded_gem_paths
 
       $LOAD_PATH.reject! do |p|
-        next if File.expand_path(p).start_with?(bundler_lib)
+        expanded_p = File.expand_path(p)
+        next if expanded_p.start_with?(bundler_lib)
+        if File.exist?(expanded_p) && File.respond_to?(:realpath)
+          next if File.realpath(expanded_p).start_with?(bundler_lib)
```

Comment:
`bundler_lib` should also be resolved with `File.realpath` (probably in `def bundler_ruby_lib` itself).
Specifically, the value of `__FILE__` is not a realpath and can contain symlinks (see https://github.com/oracle/truffleruby/issues/1383#issuecomment-403162930 for details).